veles/secrets/pgpass/pgpass_test.go

```diff
@@ -0,0 +1,168 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pgpass_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	"github.com/google/osv-scalibr/extractor/filesystem/simplefileapi"
+	scalibrfs "github.com/google/osv-scalibr/fs"
+	"github.com/google/osv-scalibr/inventory"
+	"github.com/google/osv-scalibr/veles/secrets/pgpass"
+)
+
+func TestFileRequired(t *testing.T) {
+	tests := []struct {
+		name         string
+		path         string
+		wantRequired bool
+	}{
+		{
+			name:         "valid .pgpass",
+			path:         "/foo/.pgpass",
+			wantRequired: true,
+		},
+		{
+			name:         "valid .pgpass",
+			path:         ".pgpass",
+			wantRequired: true,
+		},
+		{
+			name:         "invalid .pgpass",
+			path:         "/foo.pgpass",
+			wantRequired: false,
+		},
+		{
+			name:         "invalid .pgpass",
+			path:         "/foo.pgpass.ext",
+			wantRequired: false,
+		},
+		{
+			name:         "invalid .pgpass",
+			path:         "/foo_pgpass_zoo.ext",
+			wantRequired: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var e filesystem.Extractor = pgpass.Extractor{}
+			if got := e.FileRequired(simplefileapi.New(tt.path, nil)); got != tt.wantRequired {
+				t.Fatalf("FileRequired(%s): got %v, want %v", tt.path, got, tt.wantRequired)
+			}
+		})
+	}
+}
+
+func TestExtract(t *testing.T) {
+	tests := []struct {
+		name        string
+		path        string
+		wantSecrets []*inventory.Secret
+	}{
+		{
+			name: "valid .pgpass file",
+			path: "testdata/valid",
+			wantSecrets: []*inventory.Secret{
+				{
+					Secret: pgpass.Pgpass{Hostname: "localhost",
+						Port:     "5432",
+						Database: "mydb",
+						Username: "myuser",
+						Password: "mypassword"},
+					Location: "testdata/valid",
+				},
+				{
+					Secret: pgpass.Pgpass{Hostname: "hostname",
+						Port:     "1234",
+						Database: "testdb",
+						Username: "testuser",
+						Password: "testpass123"},
+					Location: "testdata/valid",
+				},
+				{
+					Secret: pgpass.Pgpass{Hostname: "hostname",
+						Port:     "1234",
+						Database: "testdb",
+						Username: "testuser",
+						Password: "passw*ord"},
+					Location: "testdata/valid",
+				},
+				{
+					Secret: pgpass.Pgpass{Hostname: "*",
+						Port:     "*",
+						Database: "db",
+						Username: "admin",
+						Password: "supersecret"},
+					Location: "testdata/valid",
+				},
+				{
+					Secret: pgpass.Pgpass{Hostname: "prod.example.com",
+						Port:     "5432",
+						Database: "db",
+						Username: "admin",
+						Password: `pass\:word`},
+					Location: "testdata/valid",
+				},
+			},
+		}, {
+			name:        "invalid .pgpass file",
+			path:        "testdata/invalid",
+			wantSecrets: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := pgpass.New()
+			d := t.TempDir()
```

Comment:
We have a helper lib for setting up the ScanInput for filesystem.extractor tests: extracttest.GenerateScanInputMock
example: https://github.com/google/osv-scalibr/blob/7bc99432e994668e30581245401094e4c68802d4/extractor/filesystem/language/rust/cargolock/cargolock_test.go#L170